### PR TITLE
Send cancellation email by default when cancelling an order

### LINKED ISSUE #9043: No cancellation emails are sent after cancelling an order
## Description
Since #8766 has been merged, no cancellation emails are sent after cancelling an order. This applies to cancellations by customers on account#/orders as well as cancellations by shop owner on page admin/orders/Rxyz/edit
The only way to trigger the email is to delete all items of the order on page admin/orders/Rxyz/edit and make sure the checkbox is activated to send the email when asked.

## Expected Behavior
Cancellation emails should always be sent. The only way not to send them is when deleting all items on the order edit page and then deactivating the checkbox when asked.

## Actual Behaviour
No cancellation emails are sent. The only way to send them is when deleting all items on the order edit page and then activating the checkbox when asked.

## Steps to Reproduce
1. As customer place an order
2a. As shop manager cancel the order on page admin/orders/Rxyz/edit OR 
2b. as a customer cancel the order on page account#/orders (only available if the shop allows to edit and cancel orders)
3. Observe that there is no email sent

## Animated Gif/Screenshot
No screenshot because it is an email sending issue.

## Workaround
On page admin/orders/Rxyz/edit delete all items of the order individually instead of cancelling the complete order. When asked, after deleting the last item, activate the checkbox to send the cancellation email.

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment
* Version used: 4.1.22
* Browser name and version: Firefox 98.0.2 (64-Bit)
* Operating System and version (desktop or mobile): Ubuntu 20.04 LTE Desktop

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #9043

#### What should we test?

1. As customer place an order
  2. As shop manager cancel the order on page admin/orders/Rxyz/edit OR
  3. or, as a customer cancel the order on page account#/orders (only available if the shop allows to edit and cancel orders)
4. Observe that email is sent

#### Release notes

Changelog Category: User facing changes 

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by mkllnk — APPROVED** [review]
A spec would have been good. :wink: :see_no_evil:

**Comment by jibees** [comment]
> A spec would have been good. 😉 🙈

Right, done [a511fca](https://github.com/openfoodfoundation/openfoodnetwork/pull/9045/commits/a511fcab2c0d7377713e126c7d1a631de91a44c1) 👍

File: spec/system/admin/order_spec.rb [hunk 1]
```diff
@@ -208,6 +208,22 @@ def new_order_with_distribution(distributor, order_cycle)
     end
   end
 
+  context "user can cancel an order", :debug do
```
Comment (mkllnk) [inline spec/system/admin/order_spec.rb:211]:
```suggestion
  context "user can cancel an order" do
```

**Comment by georgethoppil** [comment]
Good stuff! Quick question, when we delete the last item in the cart, we also cancel the order and we present the popup to send the email. If I remember, it is not selected right now. Do we also want to have that checked now if we are defaulting to sending the cancellation email

**Review by georgethoppil — APPROVED** [review]

**Comment by drummer83** [comment]
I have tested this PR and compared 'before' and 'after'.

To understand the current situation I staged master and cancelled a bunch of orders:
- cancelled an order as customer at another shop in account/orders --> emails sent
- cancelled an order as customer at my own shop in account/orders --> emails sent
- cancelled an order as customer at another shop in account/orders --> emails sent
- cancelled an order as customer at another shop on order confirmation screen --> emails sent
- cancelled an order as hub manager on admin/orders/R208254634/edit --> **no emails sent**
- cancelled an order as hub manager on bulk order management by deleting last item --> **no emails sent**
- cancelled an order as hub manager on admin/orders/R208254634/edit by deleting last item --> selected 'no email' --> no emails sent
- cancelled an order as hub manager on admin/orders/R208254634/edit by deleting last item --> selected 'yes, send email' --> emails sent
- cancelled an order as hub manager on admin/orders/R208254634/edit --> **no emails sent**
- cancelled an order as hub manager on admin/orders/R208254634/edit by deleting last item --> selected 'no email' --> no emails sent
- cancelled an order as customer at another shop in account/orders --> emails sent
- cancelled an order as customer at another shop on order confirmation screen --> emails sent

After that I staged the PR and tried the same again:
- cancelled an order as customer at another shop in account/orders --> emails sent
- cancelled an order as customer at another shop on order confirmation screen --> emails sent
- cancelled an order as hub manager on admin/orders/R208254634/edit --> **emails sent**
- cancelled an order as hub manager on bulk order management by deleting last item --> **emails sent**
- cancelled an order as hub manager on admin/orders/R208254634/edit by deleting last item --> selected 'no email' --> no emails sent
- cancelled an order as hub manager on admin/orders/R208254634/edit by deleting last item --> selected 'yes, send email' --> emails sent
- cancelled an order as hub manager on admin/orders/R208254634/edit --> **emails sent**
- cancelled an order as hub manager on admin/orders/R208254634/edit by deleting last item --> selected 'no email' --> no emails sent
- cancelled an order as customer at another shop in account/orders --> emails sent
- cancelled an order as customer at another shop on order confirmation screen --> emails sent

### Summary
After this PR the only cases I found where no cancellation emails are sent are the ones where I was asked if I wanted to and I selected 'no'. Exactly as I would expect it! :partying_face: 

Thanks @jibees!
Ready to go! :rocket:

**Pull request merged**